backend/battling/views.py

```diff
@@ -1,93 +1,81 @@
-from django.shortcuts import get_object_or_404, render
+from django.contrib import messages
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
 from django.urls import reverse_lazy
+from django.views.generic import CreateView, DetailView, UpdateView
 from django.views.generic.base import TemplateView
-from django.views.generic.edit import CreateView, UpdateView
 
-from battling.forms import CreatorForm, OpponentForm
-from battling.models import Battle
-from pokemon.models import Pokemon
-from services.battles import run_battle_and_send_email
+from battling.forms import CreatorForm, TeamForm
+from battling.models import Battle, Team
+
+
+# from services.battles import run_battle_and_send_email
 
 
 class Home(TemplateView):
     template_name = "battling/home.html"
 
 
-class Invite(TemplateView):
-    template_name = "battling/invite.html"
-
+class CreateBattle(CreateView):
+    model = Battle
+    form_class = CreatorForm
+    template_name = "battling/create_battle.html"
 
-class Opponent(TemplateView):  # will become an email
-    template_name = "battling/opponent.html"
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        battle = form.save()
 
+        team = Team.objects.create(battle=battle, trainer=self.request.user)
 
-class BattleEnd(TemplateView):
-    template_name = "battling/battle_end.html"
+        return HttpResponseRedirect(reverse_lazy("create_team", args=(team.id,)))
 
 
-class CreateBattle(CreateView):
-    model = Battle
-    form_class = CreatorForm
-    template_name = "battling/create_battle.html"
-    success_url = reverse_lazy("invite")
+class CreateTeam(UpdateView):
+    model = Team
+    form_class = TeamForm
+    template_name = "battling/create_team.html"
+    success_url = reverse_lazy("home")
 
     def form_valid(self, form):
-        pokemon = form.cleaned_data
+        battle = self.get_object().battle
+        battle_creator = battle.creator
+
+        # self.get_object().trainer will be replaced by self.request.user
+        # when the login feature is implemented
+        if self.get_object().trainer == battle_creator:
+            messages.success(self.request, "Your battle was created!")
 
-        form.instance.pokemon_1 = pokemon["creator_pokemon_1"]
-        form.instance.pokemon_2 = pokemon["creator_pokemon_2"]
-        form.instance.pokemon_3 = pokemon["creator_pokemon_3"]
+        else:
+            messages.success(self.request, "Battle ended! Check e-mail for results.")
 
-        form.instance.save()
+            # run_battle_and_send_email(battle)
 
         return super().form_valid(form)
 
 
-class EnterBattle(UpdateView):
-    model = Battle
-    form_class = OpponentForm
-    template_name = "battling/enter_battle.html"
-    success_url = reverse_lazy("battle_end")
+class JoinBattle(UpdateView):
```

Comment:
Hey, what's your reasoning to create this view? Some inconsistencies I see here: 
- Your view has `Team` as a model but uses the `CreatorForm` which is related to `Battle`.
- You have a form but you never save it.